sdk/cosmos/azure-cosmos-spark_3_2-12/pom.xml

```diff
@@ -448,6 +475,10 @@
                       <pattern>com.google</pattern>
                       <shadedPattern>${shadingPrefix}.com.google</shadedPattern>
                     </relocation>
+                    <relocation>
```

Comment:
Looks good on first glance - fwded you a mail with instructions to use jdeps against the shaded jar to triple-check the correctness of the shading settings.

Reply:
validated with the instructions